### ISSUE TITLE
create_key(response.request) of session.cache.responses.values() creates wrong cache id

### ISSUE THREAD

**Issue by glensc** [requests-cache/requests-cache#745]
### The problem

`create_key(response.request)` when walking `session.cache.responses.values()` creates wrong cache id that can't be used to delete.

Underlying problem seem to be that `request.body` gives `b'None'` rather than expected `''`.

See details here:
- https://github.com/Taxel/PlexTraktSync/pull/1277#issuecomment-1346201593

### Expected behavior

Able to do `session.cache.delete()` with `session.cache.responses` objects

### Steps to reproduce the behavior

1. take `response` from `session.cache.responses.values()`
2. call `delete_keys = [create_key(response.request)]`
3. try `session.cache.delete(*delete_keys)` on the keys

### Workarounds

Create new request object: `Request('GET', response.url)`

### Environment
- requests-cache version: `0.9.7`
- Python version: `3.10`
- Platform: macOS

**Comment by JWCook** [comment]
Thanks for narrowing that down. An empty request body being encoded as `b'None'` was a bug that I believe was fixed in 0.9.0 (#421). Almost all minor releases (and some patch releases) of requests-cache include some changes to request matching, which results in different cache keys for most requests. But not all requests; for example, if only request parameter matching changed, any requests without request parameters won't be affected.

That means that after updating to a newer version of requests-cache, most of your previously cached items won't be used again. Usually deleting expired responses will take care of that, unless there's a very long expiration set (or no expiration).

So if you're using requests-cache 0.9, take a response from the cache that was previously created with requests-cache 0.8, and call `create_key(response.request)`, the cache key will be different. You don't need to call `create_key()` in that case, though; there will always be a `response.cache_key` attribute available, which will be the exact key taken from the cache rather than a recreated one. For example:
```python
session.cache.delete(*[r.cache_key for r in responses_to_delete])
```

If you delete responses like this, does that fix the problem for you?
```python
session.cache.delete(expired=True, urls=[url])
```

If not, the best solution I can think of would be to update `BaseCache.delete(invalid=True)` to apply to not just serialization errors, but to any response that was cached with a different key function (e.g., where a recreated cache key is different from the stored cache key). That would make `delete()` slightly slower for a very large cache, but I believe that would solve any other problems caused by responses cached in a previous version of request-cache.

**Comment by glensc** [comment]
I went with`session.cache.delete(urls=[url])` in the merge request I had a problem (https://github.com/Taxel/PlexTraktSync/pull/1277#issuecomment-1347880066). at the time I was definitely testing with fresh cache item, but as trakt.tv has been down since then, I used some older entry and that indeed might have been from some older version. and r.cache_key was definitely missing on the freshly created entry, so therefore looked into create_key in first place.

**Comment by JWCook** [comment]
Okay, I see what happened. I should clarify: `response.cache_key` is always available when returned from a request method on `CachedSession`. It gets set in `BaseCache.get_response()`, so that wouldn't apply to direct access via `session.cache.responses`.

That is part of the public API, though, and I could see it being confusing if an attribute is available in some places but not others. I'll make a separate issue for that: #746

**Comment by JWCook** [comment]
The missing response.cache_key was fixed in main (#746), and will be included in another beta release soon.
